packages/ab-test-jsx/src/withABTest/withABTest.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react'
+import { ABTests } from '../ABTestsContext'
+import useABTests from '../useABTests'
+
+export type withABTestHoC<T extends ABTests> = <
+  TAProps extends {},
+  TBProps extends {},
+  TABTestName extends Extract<keyof T, string | number>
+>(
+  AVariantComponent: React.ComponentType<TAProps>,
+  BVariantComponent: React.ComponentType<TBProps>,
+  abTestName: TABTestName
+) => React.FC<TAProps & TBProps>
+
+const withABTest = <
+  TAProps extends {},
+  TBProps extends {},
+  TABTests extends ABTests,
+  TABTestName extends Extract<keyof TABTests, string | number>
+>(
+  AVariantComponent: React.ComponentType<TAProps>,
+  BVariantComponent: React.ComponentType<TBProps>,
+  abTestName: TABTestName
+) => {
+  type CombinedProps = TAProps & TBProps
+  const Wrapped: React.FC<CombinedProps> = React.memo(props => {
+    const { isB } = useABTests()
+    if (isB(abTestName)) {
+      return <BVariantComponent {...props} />
+    } else {
+      return <AVariantComponent {...props} />
+    }
+  })
+
+  Wrapped.displayName = `withABTest(A:${AVariantComponent.displayName}, B:${BVariantComponent.displayName})`
```

Comment:
this name will be a bit weird in DevTools, because it expects component name to be inside `( )`
![image](https://user-images.githubusercontent.com/1188327/62136270-f487d480-b30d-11e9-9a76-c58624c2a4e3.png)

Reply:
I think we can also move up this [small utils](https://github.com/matmalkowski/react-handyman/blob/9368f208f09bd7b09b98cad4fa873d9f74bb88e1/packages/feature-toggle-jsx/src/react-utils.ts#L7) and use it instead of `.displayName` here. Benefit is that you will see the name of the function component even if `displayName` property is not set, which is often the case.

Reply:
For now I just copied it over - we can make a simple package out of it though and refactor :) Please take another look and approve if everything is all right